src/inverted-index.js

```diff
@@ -81,36 +84,55 @@ class InvertedIndex {
   }
 
   /**
-   * Search method
+   * searchIndex
+   *
    * Searches key words from the files that has been uploaded
-   * @param {any} filenames
+   *
+   * @param {any} filename
    * @param {any} terms
-   * @returns {object} Documents
+   * @returns {object} results
    */
-  searchIndex(filenames, ...terms) {
-    filenames = filenames || Object.keys(this.indices);
+  searchIndex(filename, ...terms) {
+    const tempFilename = [];
+    if (typeof filename === 'string') {
+      tempFilename.push(filename);
+      filename = tempFilename;
+    }
+    if (filename) {
+      if (!this.validateFileName(filename)) {
```

Comment:
### Critical

@andela-oakinrele Why can't t this conditional be `ANDED` in line `101`?